addons/nuxeo-duoweb-authentication/src/main/java/org/nuxeo/duoweb/authentication/DuoFactorsAuthenticator.java

```diff
@@ -237,7 +237,7 @@ public NuxeoPrincipal createIdentity(String username) throws LoginException {
     protected NuxeoPrincipal validateUserIdentity() throws LoginException {
         UserManager manager = Framework.getService(UserManager.class);
         final RuntimeService runtime = Framework.getRuntime();
-        LoginPluginRegistry loginPluginManager = (LoginPluginRegistry) runtime.getComponent(LoginPluginRegistry.NAME);
+        LoginPluginRegistry loginPluginManager = runtime.getComponent(LoginPluginRegistry.NAME);
```

Comment:
i think it can be replaced by ` LoginPluginRegistry loginPluginManager = Framework.getService(LoginPluginRegistry.class);`  see [here](https://github.com/nuxeo/nuxeo/blob/acc05af20201cdc80bd7156b480a6214f0c9baa3/nuxeo-services/login/nuxeo-platform-login/src/main/resources/OSGI-INF/NXLoginModule-framework.xml#L5)